src/main/java/org/testng/internal/annotations/AnnotationHelper.java

```diff
@@ -42,7 +42,6 @@
   };
 
   private static final Class[] CONFIGURATION_CLASSES = new Class[] {
```

Comment:
Is `Class<? extends IAnnotation>[]` working?

Reply:
You mean something like this ?

`private static final Class<? extends IAnnotation> CONFIGURATION_CLASSES = new Class[] {..}` ?

I think that should work.

Reply:
Yes, the goal is to avoid the cast later.

Reply:
Here: https://github.com/cbeust/testng/pull/1555/files#diff-8157237004c7fa80cfefa18fa2651745R292

Reply:
Got it. Will take care once I am home.

Reply:
Amended commit and fixed it.